pkg/alert/commands.go

```diff
@@ -318,3 +316,187 @@ func NewDeleteAlertSettingConditionCommand() *cobra.Command {
 
 	return cmd
 }
+
+//NewGetAlertChannelsCommand creates the `alertchannels` command
+func NewGetAlertChannelsCommand() *cobra.Command {
+	var (
+		page         int
+		pageSize     int
+		sortField    string
+		sortOrder    string
+		templateName string
+		channelName  string
+		details      bool
+	)
+
+	cmd := &cobra.Command{
+		Use:              "alertchannels",
+		Short:            "Print the registered alert channels",
+		Example:          `alertchannels --page=1 --pageSize=10 --sortField="name" --sortOrder="asc" --templateName="test" --channelName="slack" --details`,
+		TraverseChildren: true,
+		SilenceErrors:    true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				alertchannels            api.AlertChannelResponse
+				alertchannelsWithDetails api.AlertChannelResponseWithDetails
+				err                      error
+			)
+			if details {
+				alertchannelsWithDetails, err = config.Client.GetAlertChannelsWithDetails(page, pageSize, sortField, sortOrder, templateName, channelName)
+			} else {
+				alertchannels, err = config.Client.GetAlertChannels(page, pageSize, sortField, sortOrder, templateName, channelName)
+			}
+			if err != nil {
+				golog.Errorf("Failed to retrieve alert channels. [%s]", err.Error())
+				return err
+			}
+
+			if !details {
+				return bite.PrintObject(cmd, alertchannels)
+			}
+			return bite.PrintObject(cmd, alertchannelsWithDetails.Values)
+		},
+	}
+
+	cmd.Flags().IntVar(&page, "page", 1, "--page=1")
+	cmd.Flags().IntVar(&pageSize, "pageSize", 10, "--pageSize=10")
+	cmd.Flags().StringVar(&sortField, "sortField", "", `--sortField="name"`)
+	cmd.Flags().StringVar(&sortOrder, "sortOrder", "", `--sortOrder="asc"`)
+	cmd.Flags().StringVar(&templateName, "templateName", "", `--templateName="test"`)
+	cmd.Flags().StringVar(&channelName, "channelName", "", `--channelName="slack"`)
+	cmd.Flags().BoolVar(&details, "details", false, `--details`)
+
+	bite.CanBeSilent(cmd)
+	bite.CanPrintJSON(cmd)
+
+	cmd.AddCommand(NewDeleteAlertChannelCommand())
+	cmd.AddCommand(NewCreateAlertChannelCommand())
+	cmd.AddCommand(NewUpdateAlertChannelCommand())
+
+	return cmd
+}
+
+//NewDeleteAlertChannelCommand creates `alertchannels delete` command
+func NewDeleteAlertChannelCommand() *cobra.Command {
+	var (
+		channelID string
+	)
+
+	cmd := &cobra.Command{
+		Use:              "delete",
+		Short:            "Delete an alert channel",
+		Example:          `alertchannels delete --channelID="fa0e9b96-1048-4f4c-b776-4e96ca62f37d"`,
+		TraverseChildren: true,
+		SilenceErrors:    true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := config.Client.DeleteAlertChannel(channelID)
+			if err != nil {
+				golog.Errorf("Failed to delete alert channel [%s]. [%s]", channelID, err.Error())
+				return err
+			}
+			return bite.PrintInfo(cmd, "Alert channel [%s] deleted", channelID)
+		},
+	}
+
+	cmd.Flags().StringVar(&channelID, "channelID", "", "Channel ID")
```

Comment:
is there a way to add comments here? I mean what is the channel ID in the CLI when I am hitting help for example?